sanic/handlers.py

```diff
@@ -105,7 +105,14 @@ def log(self, message, level='error'):
         getattr(logger, level)(message)
 
     def default(self, request, exception):
-        self.log(format_exc())
+        try:
+            url = repr(request.url)
+        except AttributeError:
+            url = "unknown"
```

Comment:
this may be a personal preference so I'll leave it to the committer but I think `repr(getattr(request, 'url', "unknown"))` is a bit more concise when there's a logical default. EFAP seems technically more "pythonic" but I always found it kinda verbose🤷‍♂️

Reply:
I actually prefer `repr(getattr(request, 'url', "unknown"))` too, and in this case might be slightly faster than `try/except`.

Reply:
I'm fine with both, but `try/except` should be faster than `repr(getattr(request, 'url', "unknown"))`. And in case of "url" exists, `try/except` is significantly faster than the other one.

Reply:
Your suggestion is not logically equivalent to this code. In my code it sets `url = "unknown"` whereas in yours it would set `url = repr("unknown")`

Reply:
good point! actually not entirely sure why you are using repr in the pr, `request.url` is a plain string I thought?

Reply:
@yunstanford did some baseline benchmarking 

```
In [7]: a.url
Out[7]: 'http://127.0.0.1:42101/'

In [8]: %%timeit
   ...: try:
   ...:     url = a.url
   ...: except AttributeError:
   ...:     url = "unknown"
   ...:
5.51 µs ± 250 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

Reply:
Because the last thing I want to have at this point is to crash because of some stupid encoding problem with the URL.